packages/auth-client/src/auth-client.js

```diff
@@ -380,6 +380,28 @@ class AuthClient {
   _removeCache(wallet) {
     window.localStorage.removeItem(`auth:${wallet}`)
   }
+
+  /**
+   * Get server time
+   */
+  async getServerTime() {
+    try {
+      let url = new URL(this.authServer)
+      url.pathname = '/api/now'
+      url = url.toString()
+
+      const rawResponse = await fetch(url)
+
+      const response = await rawResponse.json()
+
+      return response.body.timestamp || Date.now()
```

Comment:
If body contents of the request can be retrieved but timestamp field is missing, I would rather see this throwing an exception and at least logging the error, and then returning client side Date.now(). Instead of silently failing

Reply:
Thanks @sparrowDom. Have made that change